test/extended/builds/cluster_config.go

```diff
@@ -0,0 +1,99 @@
+package builds
+
+import (
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Feature:Builds][Serial][Slow] alter builds via cluster configuration", func() {
+	defer g.GinkgoRecover()
+	var (
+		buildFixture           = exutil.FixturePath("testdata", "builds", "test-build.yaml")
+		defaultConfigFixture   = exutil.FixturePath("testdata", "builds", "cluster-config.yaml")
+		blacklistConfigFixture = exutil.FixturePath("testdata", "builds", "cluster-config", "registry-blacklist.yaml")
+		whitelistConfigFixture = exutil.FixturePath("testdata", "builds", "cluster-config", "registry-whitelist.yaml")
+		oc                     = exutil.NewCLI("build-cluster-config", exutil.KubeConfigPath())
+	)
+
+	g.Context("", func() {
+
+		g.BeforeEach(func() {
+			exutil.PreTestDump()
+		})
+
+		g.JustBeforeEach(func() {
+			g.By("waiting for default service account")
+			err := exutil.WaitForServiceAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()), "default")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By("waiting for builder service account")
+			err = exutil.WaitForServiceAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()), "builder")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			oc.Run("create").Args("-f", buildFixture).Execute()
+		})
+
+		g.AfterEach(func() {
+			if g.CurrentGinkgoTestDescription().Failed {
+				exutil.DumpPodStates(oc)
+				exutil.DumpPodLogsStartingWith("", oc)
+				exutil.DumpConfigMapStates(oc)
+			}
+			oc.AsAdmin().Run("apply").Args("-f", defaultConfigFixture).Execute()
+		})
+
+		g.Context("registries config context", func() {
+
+			g.It("should default registry search to docker.io for image pulls", func() {
+				g.By("apply default cluster configuration")
+				err := oc.AsAdmin().Run("apply").Args("-f", defaultConfigFixture).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				g.By("waiting 1s for build controller configuration to propagate")
+				time.Sleep(1 * time.Second)
+				g.By("starting build sample-build and waiting for success")
+				br, err := exutil.StartBuildAndWait(oc, "sample-build")
+				o.Expect(err).NotTo(o.HaveOccurred())
+				br.AssertSuccess()
+				g.By("expecting the build logs to indicate docker.io is the search registry")
+				buildLog, err := br.LogsNoTimestamp()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(buildLog).To(o.ContainSubstring("defaulting registry to docker.io"))
+			})
```

Comment:
missing a test that confirms the admin can control the registry defaulting?

make it default to quay.io and confirm the build tries to pull from there (the build will fail of course but that's fine).